src/mixin/entity-mixin-lit.js

```diff
@@ -18,60 +18,27 @@ export const EntityMixinLit = superclass => class extends superclass {
 			/**
 			 * Entity object that extends the Entity class.
 			 */
-			_entity: { type: Object },
+			entity: { type: Object }
 		};
 	}
 
-	constructor() {
-		super();
-	}
-
-	set href(href) {
-		const oldValue = this.__href;
-		this.__href = href;
-		this.requestUpdate('href', oldValue);
-	}
-
-	get href() {
-		return this.__href;
-	}
-
-	set token(token) {
-		const oldValue = this.__token;
-		this.__token = token;
-		this.requestUpdate('token', oldValue);
-	}
-
-	get token() {
-		return this.__token;
-	}
-
-	set _entity(entity) {
-		this.__entity = entity;
-	}
-
-	get _entity() {
-		return this.__entity;
-	}
-
 	disconnectedCallback() {
 		super.disconnectedCallback();
 		// this calls this._entity.dispose() if entity is actually an entity.
 		// Note this will dispose all child entities used.
-		dispose(this._entity);
+		dispose(this.entity);
 	}
 
-	requestUpdate(propChanges, oldValue) {
```

Comment:
I was on the fence about where best to put our hook to load the entity too.
I went with shouldUpdate because I wanted to prevent renders when we don't have a token or an href yet, and it seemed simpler to just keep the logic all together.

Reply:
Something to consider is d2lfetch-auth is almost always in the middleware chain. https://github.com/Brightspace/d2l-fetch-auth/blob/master/src/unframed/d2lfetch-auth.js.  Tokens are semi optional

Reply:
Yeh, that's a good point. I'm not clear on the best way to handle that though. As if the href is set before the token, but you really intend to set a token, it can end up making the request without the token.